api/metrics/v1/http.go

```diff
@@ -274,5 +277,25 @@ func NewHandler(read, write *url.URL, upstreamCA []byte, opts ...HandlerOption)
 		})
 	}
 
+	if rulesEndpoint != nil {
+		client, err := rules.NewClient(rulesEndpoint.String())
+		if err != nil {
+			level.Warn(c.logger).Log("msg", "could not create rules endpoint client")
+			return r
+		}
+
+		rh := rulesHandler{client: client}
+
+		r.Group(func(r chi.Router) {
+			r.Use(c.uiMiddlewares...)
+			r.Get(rulesRoute, rh.get)
```

Comment:
This route is same as the [Rules endpoint in Prom](https://prometheus.io/docs/prometheus/latest/querying/api/#rules) (also exposed by Thanos Querier) so it'll shadow it breaking anything (like Thanos Rules UI) that depends on it. I think we should change the `rulesRoute` to avoid confusion.

Reply:
very good point - I've changed it (now I think is in line with the [proposal](https://observatorium.io/docs/proposals/20201019-prometheus-rules-in-observatorium-api.md/) too)

Reply:
changed it to `api/v1/rulesfile` since either `api/v1/rules` and `/rules` would still create conflict